packages/grafana-ui/src/components/Table/utils.ts

```diff
@@ -299,21 +323,41 @@ function toNumber(value: any): number {
 }
 
 export function getFooterItems(
-  filterFields: Array<{ field: Field }>,
+  filterFields: Array<{ id: string; field: Field }>,
   values: any[number],
   options: TableFooterCalc,
   theme2: GrafanaTheme2
 ): FooterItem[] {
-  return filterFields.map((data, i) => {
+  // Deep clone the `filterFields` array, since we will be conditionally mutating it later on.
+  let footerFieldData: Array<{ id: string; field: Field }> = cloneDeep(filterFields);
```

Comment:
I'm wondering whether we can avoid using `cloneDeep` in here since it could be quite expensive in some cases. Could we not just form a new array for the map below in a different way? Maybe something like `[ fieldToAdd, ...filterFields]`?